mParticle-OneTrust/MPKitOneTrust.m

```diff
@@ -55,6 +73,58 @@ - (void)start {
     });
 }
 
+// Listener for Consent Change dispatched by One Trust
+-(void)actionConsent:(NSNotification*)notification {
+    NSString *category = notification.name; // Cookie Name
+    NSNumber *status = notification.object; // BOOL-ish value for consent
+
+    // Fire consent change event
+    [self createConsentEvent:category :status];
+}
+
+// Parses the raw consent mapping from the mParticle UI into simple map
+- (NSDictionary*)parseConsentMapping:(NSString*)rawConsentMapping {
+    NSData *jsonData = [rawConsentMapping dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error;
+
+    NSMutableDictionary *consentMapping = [[NSMutableDictionary alloc] init];
+
+    id jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
+
+        if (error) {
+            NSLog(@"Error parsing JSON: %@", error);
+        }
+        else
+        {
+            if ([jsonObject isKindOfClass:[NSArray class]])
+            {
+                for (NSDictionary *element in jsonObject) {
+                    consentMapping[element[@"value"]] = element[@"map"];
```

Comment:
I would wrap this set in a check for `if (element[@"value"] != nil && element[@"map] != nil)` so you avoid the risk of potentially setting e.g. a nil value into the dictionary which can cause a crash.